### ISSUE TITLE
Dropout scaling

### ISSUE THREAD

**Issue by BaldwinTheThird** [fchollet/keras#2923]
How exactly is Dropout handled in Keras? I can see in the code that inputs are multiplied with matrices that set some inputs equal to zero which makes a lot of sense. Dropout also requires scaling down the final weights of the trained model depending on what dropout fraction was used. Where exactly does this happen?

**Comment by ChristianThomae** [comment]
Keras does both operations (set to zero and inverse scaling) at train time:
https://github.com/fchollet/keras/blob/master/keras/backend/theano_backend.py#L788-L789
